core/src/main/java/org/elasticsearch/rest/action/cat/RestTable.java

```diff
@@ -118,20 +118,26 @@ public static RestResponse buildTextPlainResponse(Table table, RestChannel chann
         String[] columnOrdering = request.paramAsStringArray("s", null);
 
         List<Integer> rowOrder = new ArrayList<>();
-        for(int i = 0; i < table.getRows().size(); i++) {
+        for (int i = 0; i < table.getRows().size(); i++) {
             rowOrder.add(i);
         }
 
-        if(columnOrdering != null) {
+        if (columnOrdering != null) {
             Map<String, String> headerAliasMap = table.getAliasMap();
             List<ColumnOrderElement> ordering = new ArrayList<>();
-            for(int i = 0; i < columnOrdering.length; i++) {
-                String[] orderingEntry = columnOrdering[i].split(":");
-                boolean reverse = (orderingEntry.length > 1 && orderingEntry[1].equals("desc"));
-                if(headerAliasMap.containsKey(orderingEntry[0])) {
-                        ordering.add(new ColumnOrderElement(headerAliasMap.get(orderingEntry[0]), reverse));
+            for (int i = 0; i < columnOrdering.length; i++) {
+                String columnHeader = columnOrdering[i];
+                boolean reverse = false;
+                if (columnHeader.length() > 5 && columnHeader.endsWith(":desc")) {
```

Comment:
I'm not sure the length comparison buys you much. I'm fairly sure `String#endsWith` already checks that super duper early.